app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,53 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :set_post, only: [:show, :update, :destroy]
+
+  def index
+    @posts = Post.all
+
+    # render json: @posts
+    render_resource(@posts)
+  end
+
+  def show
+    render json: @post
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    # if @post.save
+    #   render json: @post, status: :created, location: @post
+    # else
+    #   render json: @post.errors, status: :unprocessable_entity
+    # end
+    render_resource(@post) if @post.save
```

Comment:
это и далее все прекрасно. В Show вроде тоже можно так сделать. Проверь только =)

Reply:
А, нет.
```suggestion
    @post = Post.create(post_params)
    render_resource(@post)
```
Ну и тоже понятное дело проверь. Сейчас у тебя получается, что если ошибка то оно ничего не срендерит вообще.